### PR TITLE
Implement expansion tests

### PR DESCRIPTION
**Pull request by eupn**
Related to #1117.

This PR implements serde's `#[derive(Serialize, Deserialize)]` macros expansion tests using [macrotest](https://crates.io/crates/macrotest).

~To pass CI it requires `cargo expand` and `rustfmt` to be available.~

### CODE REVIEW

File: test_suite/tests/expand/derive/enum/de_enum.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use serde::Deserialize;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
```
Comment (dtolnay) [inline test_suite/tests/expand/derive/enum/de_enum.rs]: We should not be testing the compiler's Debug and PartialEq derives here. I don't want broken tests when the compiler changes those...

File: test_suite/tests/expand/derive/enum/de_enum.expanded.rs [hunk 2]
```diff
@@ -0,0 +1,1373 @@
...
+                                        _serde::export::Ok(__val) => __val,
+                                        _serde::export::Err(__err) => {
+                                            return _serde::export::Err(__err);
+                                        }
+                                    } {
+                                        _serde::export::Some(__value) => __value,
+                                        _serde::export::None => {
+                                            return _serde::export::Err(
+                                                _serde::de::Error::invalid_length(
+
```
Comment (dtolnay) [inline test_suite/tests/expand/derive/enum/de_enum.expanded.rs]: Could you make sure these files are written as proper text files with a newline character at the end of every line?

File: test_suite/tests/expandtest.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#[cfg(not(target_os = "emscripten"))]
+#[rustversion::attr(not(nightly), ignore)]
```
Comment (dtolnay) [inline test_suite/tests/expandtest.rs]: Would it be possible to also #\[ignore\] if cargo-expand is not installed? We can add a build.rs to the test_suite crate to detect cargo-expand.

File: test_suite/tests/expand/derive/enum/de_enum.rs (hunk 1 above)
Comment (eupn) [reply test_suite/tests/expand/derive/enum/de_enum.rs]: Thanks, I forgot to remove these.

File: test_suite/tests/expand/derive/enum/de_enum.expanded.rs (hunk 2 above)
Comment (eupn) [reply test_suite/tests/expand/derive/enum/de_enum.expanded.rs]: This is an issue that require modification to `macrotest`, I filed it here: https://github.com/eupn/macrotest/issues/33

File: test_suite/tests/expandtest.rs (hunk 3 above)
Comment (eupn) [reply test_suite/tests/expandtest.rs]: Yes, I'll try.

Comment (dtolnay) [reply test_suite/tests/expandtest.rs]: Sounds good. Later it could be its own crate: https://github.com/dtolnay/request-for-implementation/issues/44

Comment (eupn) [reply test_suite/tests/expandtest.rs]: Done.

File: test_suite/tests/expand/derive/enum/de_enum.expanded.rs (hunk 2 above)
Comment (eupn) [reply test_suite/tests/expand/derive/enum/de_enum.expanded.rs]: Done, updated the crate.

**Comment by eupn** [comment]
@dtolnay ping

**Pull request merged**